lib/portage/process.py

```diff
@@ -241,6 +245,37 @@ def cleanup():
     pass
 
 
+@dataclass(frozen=True)
+class EnvStats:
+    env_size: int
+    env_largest_name: str
+    env_largest_size: int
+
+
+def calc_env_stats(env) -> EnvStats:
+    @lru_cache(1024)
+    def encoded_length(s):
+        return len(os.fsencode(s))
+
+    env_size = 0
+    env_largest_name = None
+    env_largest_size = 0
+    for env_name, env_value in env.items():
+        env_name_size = encoded_length(env_name)
+        env_value_size = encoded_length(env_value)
+        # Add two for '=' and the terminating null byte.
+        total_size = env_name_size + env_value_size + 2
+        if total_size > env_largest_size:
+            env_largest_name = env_name
+            env_largest_size = total_size
+        env_size += total_size
+
+    return EnvStats(env_size, env_largest_name, env_largest_size)
+
+
+env_to_large_warnings = 0
```

Comment:
Typo:
```suggestion
env_too_large_warnings = 0
```
(and propagate the fixed spelling)

Reply:
fixed